src/ucp/tag/eager_multi.c

```diff
@@ -219,11 +220,12 @@ void ucp_proto_eager_sync_ack_handler(ucp_worker_h worker,
 {
     ucp_request_t *req;
 
-    UCP_WORKER_EXTRACT_REQUEST_BY_ID(&req, worker, rep_hdr->req_id, return,
-                                     "EAGER_S ACK %p", rep_hdr);
+    UCP_REQUEST_GET_BY_ID(&req, worker, rep_hdr->req_id, 0, return,
```

Comment:
why not extract here?

Reply:
we should delete request ID after all requests are sent
otherwise - getting local request ID will be broken, since it is set to `INVALID`

Reply:
we are packing local request id only for first frag

Reply:
we always fills AM first hdr with request ID here:
`ucp_tag_eager_sync_zcopy_multi()`
event if we will send it as the middle segment

Reply:
can we fill it only if `req->send.state.dt.offset == 0`?

Reply:
yes, we can
done